torchinductor/lowering.py

```diff
@@ -2522,6 +2540,36 @@ def copy_(dst, src, non_blocking=False):
     return mutate_to(dst, src)
 
 
+@make_pointwise
+def floordiv(a, b):
+    return ops.floordiv(a, b)
+
+
+@make_pointwise
+def truncdiv(a, b):
+    return ops.truncdiv(a, b)
+
+
+@register_lowering(aten.div.Tensor_mode)
+def div_mode(a, b, rounding_mode=None):
+    both_integer = is_integer_type(a) and is_integer_type(b)
+    on_triton = is_triton_device(a) or is_triton_device(b)
+
+    # floordiv and truncdiv need special handling for integer tensors on Triton,
+    # see the discussion at https://github.com/openai/triton/issues/605
+    if rounding_mode == "floor":
+        return floordiv(a, b) if (both_integer and on_triton) else floor(div_mode(a, b))
+    if rounding_mode == "trunc":
+        return truncdiv(a, b) if (both_integer and on_triton) else trunc(div_mode(a, b))
+    if both_integer:
+        # truediv produces a float tensor even if both operands are integer types
+        return div(
+            to_dtype(a, torch.float32) if isinstance(a, TensorBox) else a,
```

Comment:
float32 should not be hardcoded here, it's `torch.get_default_dtype()`, users can set it to fp64